docs/guides/advanced/deepkit-type.md

```diff
@@ -0,0 +1,96 @@
+---
+id: using-with-deepkit-type
+title: 'Using with @deepkit/type'
+---
+
+This guide shows how to use `typegoose` with `@deepkit/type`.
+
+## Implementation
+
+Suppose you have this `Account` class decorated with `@deepkit/type`:
+
+```ts
+import { classToPlain, t, plainToClass } from '@deepkit/type';
+import { getModelForClass, prop } from '../../src/typegoose';
+
+enum Group {
+  confidential = 'confidential',
+  public = 'public',
+}
+
+class Account {
+  @t.mongoId.group(Group.public)
+  public _id: string;
+
+  @t.group(Group.public)
+  public __v: number;
+
+  @t.group(Group.public)
+  @prop()
+  public email: string;
+
+  @t.group(Group.confidential)
+  @prop()
+  public confidentialProperty?: string;
+}
+
+const AccountModel = getModelForClass(Account);
+```
+
+You can then use, for example:
+
+* `lean()` on the query:
+
+  ```ts
+  const pojo = await AccountModel.findById(id).orFail().lean().exec();
+  // groupsExclude option excludes the property group
+  const access = { groupsExclude: [Group.confidential] };
+  // deserialize pojo back to an Account instance
+  const deserialized = plainToClass(Account, pojo, access);
+ 
+  expect(deserialized).toStrictEqual(accountClassObject);
+  ```
+
+* or a normal document:
+
+  ```ts
+  // exec returns a Mongoose Object
+  const doc = await AccountModel.findById(id).orFail().exec();
+  const access = { groupsExclude: [Group.confidential] };
+  // serialize Account instance back to a Plain Old Javascript Object
+  const serialized = classToPlain(Account, doc, access);
+  ```
+
+As you can see from these code examples, there is:
+
+* an `access` or "grouping" definition needed to ouput the required properties for any serializing or deserializing. The `@deepkit/type` author suggests not using groupings at all, as they cause `@deepkit/type`'s performance to deteriorate.
```

Comment:
do you maybe have a link to where this is stated?

Reply:
https://deepkit.io/documentation/type/serialization#groups

Reply:
thanks, could you maybe add that link to the docs (to provide a source on that statement)?

Reply:
@hasezoey - As I mentioned above, I'm a noob to Git and resolving conflicts and what to do to not zap your changes. VS-Code is showing me the changes and to select from them. I need to go through 8 different files. I can try this, but with no guarantee of it working properly. 

Is there an easier and safer way to be sure any changes I now make don't screw things up royally? I'll be honest too. I use Gitkraken and the options they give are like chinese to me. 

Pull - Fetch all 
Pull - Fast forward if possible
Pull - Fast forward only
Pull - Rebase

🤷 

I did what was defaulted too, Pull - Fast forward if possible, and now I have these conflicts. 😢 

Scott

Reply:
if that are the only options presented to you (and no Pull - Force), then you will have to go to the directory and manually execute git there, as stated in https://github.com/typegoose/typegoose/pull/602#discussion_r701736086

btw, `Pull - Fast forward` only works when the history did not change between your remote and local branch, but in this case *it did change*, also `Pull - Rebase` would not work, because yes, it would get the remote branch *but* it will try to also apply the old versions of you local branch again ontop of the remote branch - which most probably will lead to conflicts (or at the least, duplicated commits)

Reply:
added with fa5fa8e258609a5fd8abc43f8ef95afb9e3cb5da